sdcm/tester.py

```diff
@@ -974,21 +973,24 @@ def set_system_auth_rf(self, db_cluster=None):
         if self.test_config.REUSE_CLUSTER:
             self.log.debug("Skipping change rf of system_auth for reusing cluster")
             return
-        node = self.db_cluster.nodes[0]
+        self.set_ks_rf(keyspace="system_auth", db_cluster=db_cluster)
+
+    def set_ks_rf(self, keyspace, db_cluster=None):
+        db_cluster = db_cluster or self.db_cluster
+        node = random.choice([node for node in self.db_cluster.nodes if not node.running_nemesis])
         nodes_by_region = self.db_cluster.nodes_by_region()
         dc_by_region = self.db_cluster.get_datacenter_name_per_region(db_cluster.nodes)
         datacenters = {}
         for region in nodes_by_region:
             datacenters.update({dc_by_region[region]: len(nodes_by_region[region])})
         self.log.debug("Number of nodes by datacenter %s", datacenters)
-        NetworkTopologyReplicationStrategy(**datacenters).apply(node, "system_auth")
-        res = node.run_cqlsh('DESC KEYSPACE system_auth',
-                             num_retry_on_failure=3)
-        self.log.debug("system_auth description: %s", res.stdout)
-        self.log.info('repair system_auth keyspace ...')
+        NetworkTopologyReplicationStrategy(**datacenters).apply(node, keyspace)
+        res = node.run_cqlsh(f'DESC KEYSPACE "{keyspace}"', num_retry_on_failure=3)
+        self.log.debug("%s description: %s", keyspace, res.stdout)
+        self.log.info('repair %s keyspace ...', keyspace)
         for node in self.db_cluster.nodes:
-            node.run_nodetool(sub_cmd="repair -pr system_auth", timeout=MINUTE_IN_SEC * 20)
-        self.log.info('repair system_auth keyspace done')
+            node.run_nodetool(sub_cmd=f"repair -pr {keyspace}", timeout=MINUTE_IN_SEC * 20)
```

Comment:
I'm not sure  repair is the side effect you want in restore nemesis, isn't that part of the restore itself ?

Reply:
It happens automatically after a restore, but you do need to repair a keyspace after altering it, do you not?
Also, I don't see why that would be an issue. It's not like we're repairing after restoring the data, only the schema.

Reply:
for system_auth we are doing tha to make sure the auth data is on all nodes.

in this case isn't restore procedure puts date into each node ? i.e. if doing repair is part of the instructions for restore ? 
what the end user is going to do, and again where is the documentation for this flow, of data restored into different DC layout

Reply:
Added an option to not repair after the alteration, and used it in my function call.
Also opened an issue for the docs to be amended: https://github.com/scylladb/scylla-manager/issues/3719